safenode/src/client/api.rs

```diff
@@ -204,7 +203,21 @@ impl Client {
     pub(crate) async fn send_to_closest(&self, request: Request) -> Result<Vec<Result<Response>>> {
         let responses = self
             .network
-            .client_send_to_closest(&request)
+            .client_send_to_closest(&request, true)
+            .await?
+            .into_iter()
+            .map(|res| res.map_err(Error::Network))
+            .collect_vec();
+        Ok(responses)
+    }
+
+    pub(crate) async fn send_and_wait_till_first_rsp(
+        &self,
+        request: Request,
+    ) -> Result<Vec<Result<Response>>> {
```

Comment:
Just a simple one, but can we return a single response instead of a vec here?

Reply:
there is chance some peer may response with error.

Reply:
If I'm not wrong, the underlying `Network::send_and_get_responses()` returns the **first successful response** if `expect_all_responses=false` or will return `vec<error>` if every node returned an error.

If we don't care just about **all the errors**, then maybe we can dump the `Vec` here?